app/models/manageiq/providers/azure/manager_mixin.rb

```diff
@@ -25,6 +26,18 @@ def raw_connect(client_id, client_key, azure_tenant_id, subscription, proxy_uri
         $azure_log.warn("No region selected. Validating credentials against public environment.")
       end
 
+      endpoint_url = endpoint.respond_to?(:url) ? endpoint.url : endpoint.to_s
+
+      if endpoint_url.present?
+        begin
+          environment = ::Azure::Armrest::Environment.discover(:url => endpoint_url, :proxy => proxy_uri)
+        rescue SocketError
+          raise MiqException::MiqInvalidCredentialsError, _("Invalid endpoint")
+        end
```

Comment:
Is it always a credentials error?

Reply:
It basically means the endpoint is invalid, which I'm counting as "credentials" for our purposes. I actually couldn't find a better error, but I'm open to suggestions.

Reply:
Maybe this should be switched to an `MiqUnreachableError`.

Reply:
Agreed.